contractcourt/breach_resolver.go

```diff
@@ -0,0 +1,99 @@
+package contractcourt
+
+import (
+	"encoding/binary"
+	"io"
+
+	"github.com/lightningnetwork/lnd/channeldb"
+)
+
+// breachResolver is a resolver that will handle breached closes. In the
+// future, this will likely take over the duties the current breacharbiter has.
+type breachResolver struct {
+	// resolved reflects if the contract has been fully resolved or not.
+	resolved bool
+
+	contractResolverKit
+}
+
+// newBreachResolver instantiates a new breach resolver.
+func newBreachResolver(resCfg ResolverConfig) *breachResolver {
+	r := &breachResolver{
+		contractResolverKit: *newContractResolverKit(resCfg),
+	}
+
+	r.initLogger(r)
+
+	return r
+}
+
+// ResolverKey returns the unique identifier for this resolver.
+func (b *breachResolver) ResolverKey() []byte {
+	key := newResolverID(b.ChanPoint)
+	return key[:]
+}
+
+// Resolve queries the breacharbiter to see if the justice transaction has been
+// broadcast.
+func (b *breachResolver) Resolve() (ContractResolver, error) {
+	breached, err := b.IsBreached(&b.ChanPoint)
+	if err != nil {
+		return nil, err
+	}
+
+	// If the ChanPoint is still in the breacharbiter's database, then the
+	// final justice transaction hasn't confirmed on-chain.
+	if breached {
+		return nil, nil
+	}
+
+	// Otherwise, breached is false and the last justice transaction is on
+	// chain. Since the breacharbiter is notified of the breach first and
+	// then the channel arbitrator is notified, this is not racy code.
+	b.resolved = true
+
+	// Checkpoint the resolved state.
+	return nil, b.Checkpoint(b)
+}
+
+// Stop signals the breachResolver to stop. This is not necessary to implement
+// as the breachResolver makes one function call and returns.
+func (b *breachResolver) Stop() {
+}
+
+// IsResolved returns true if the breachResolver is fully resolved and cleanup
+// can occur.
+func (b *breachResolver) IsResolved() bool {
+	return b.resolved
```

Comment:
I think we maybe want to swap the logic here with what's currently in `Resolve`?

Reply:
This follows the same pattern of the other resolvers, wdym?

Reply:
Clarified in the other comment, marking this one as being resolved. This'll need some light synchronization to avoid a race condition.

Reply:
I don't think this can be a race condition since `IsResolved` and `Resolve` are called from the same thread?